### PR TITLE
[Feature] Add an e2e test for Autoscaler to scale up by manually updating `minReplicas`

### LINKED ISSUE #2576: [Feature] Add an e2e test for Autoscaler to scale up by manually updating `minReplicas`
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/kuberay/issues) and found no similar feature requirement.

### Description

Users may manually update `minReplicas` to a value greater than `replicas`. Then, KubeRay will create new Pods to ensure the worker group has `$minReplicas` Pods. It's the Autoscaler's duty to update `replicas` to ensure it is equal to or greater than `minReplicas`.

For this test,

* Create a RayCluster with the Autoscaler enabled, containing a worker group with `replicas: 1` and `minReplicas: 0`.
* Update `minReplicas` from 0 to 2.
* KubeRay should create an additional Pod.
* Submit some Ray detached actors to trigger autoscaling to 5 Pods.
* Verify whether the Autoscaler successfully scales up to 5 Pods.
* Check whether `replicas` is set to 5.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by LeoLiao123**
## Why are these changes needed?

See #2576 

## Related issue number

Closes #2576 

## Test Log
This test follows these sequential steps:
* Create a RayCluster with the Autoscaler enabled, containing a worker group with replicas: 1 and minReplicas: 0.

```
{"msg": "reconcilePods", "desired workerReplicas": 1, "maxReplicas": 5, "minReplicas": 0, "replicas": 1}
```

* Update minReplicas from 0 to 2.
* KubeRay should create an additional Pod.

```
{"msg": "reconcilePods", "desired workerReplicas": 2, "maxReplicas": 5, "minReplicas": 2, "replicas": 1}
{"msg": "Number workers to add": 1, "Worker group": "test-group"}
{"msg": "Created worker Pod for RayCluster"}
```

* Submit some Ray detached actors to trigger autoscaling to 5 Pods.

```
"Executing command: [python /home/ray/test_scripts/create_detached_actor.py actor0]"
"Executing command: [python /home/ray/test_scripts/create_detached_actor.py actor1]"
"Executing command: [python /home/ray/test_scripts/create_detached_actor.py actor2]"
"Executing command: [python /home/ray/test_scripts/create_detached_actor.py actor3]"
"Executing command: [python /home/ray/test_scripts/create_detached_actor.py actor4]"
```

* Verify whether the Autoscaler successfully scales up to 5 Pods.

```
{"msg": "reconcilePods", "workerReplicas": 5, "NumOfHosts": 1, "runningPods": 2, "diff": 3}
{"msg": "Number workers to add": 3, "Worker group": "test-group"}
```

* Check whether replicas is set to 5.

```
"new status": {
    "desiredWorkerReplicas": 5,
    "minWorkerReplicas": 2,
    "maxWorkerReplicas": 5,
    "observedGeneration": 3
}
```

## Screenshot
image

## Checks
- [x] I've made sure the tests are passing.
- Testing Strategy
   - [x] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by LeoLiao123** [comment]
@MortalHappiness @kevin85421  PTAL

**Comment by kevin85421** [comment]
cc @MortalHappiness can you review this PR?

File: ray-operator/test/e2eautoscaler/support.go [hunk 1]
```diff
@@ -130,3 +131,11 @@ func workerPodTemplateApplyConfiguration() *corev1ac.PodTemplateSpecApplyConfigu
 						corev1.ResourceMemory: resource.MustParse("1G"),
 					}))))
 }
+
+func getRayClusterWorkerReplicas(cluster *rayv1.RayCluster) int32 {
```
Comment (MortalHappiness) [inline ray-operator/test/e2eautoscaler/support.go:135]: Move this function to `ray-operator/test/support/ray.go` and rename it to `GetRayClusterWorkerGroupReplicaSum`

File: ray-operator/test/e2eautoscaler/raycluster_autoscaler_test.go [hunk 2]
```diff
@@ -253,3 +257,75 @@ func TestRayClusterAutoscalerWithDesiredState(t *testing.T) {
...
+func TestRayClusterAutoscalerMinReplicasUpdate(t *testing.T) {
+	test := With(t)
+	g := gomega.NewWithT(t)
+
+	// Create a namespace
+	namespace := test.NewTestNamespace()
+	test.StreamKubeRayOperatorLogs()
+
+	// Scripts for creating and terminating detached actors to trigger autoscaling
+	scriptsAC := newConfigMap(namespace.Name, files(test, "create_detached_actor.py", "terminate_detached_actor.py"))
```
Comment (MortalHappiness) [inline ray-operator/test/e2eautoscaler/raycluster_autoscaler_test.go:270]: `terminate_detached_actor.py` is not used in this test, remove it.

**Review by MortalHappiness — CHANGES_REQUESTED** [review]

File: ray-operator/test/e2eautoscaler/support.go (hunk 1 above)
Comment (LeoLiao123) [reply ray-operator/test/e2eautoscaler/support.go]: Done

File: ray-operator/test/e2eautoscaler/raycluster_autoscaler_test.go (hunk 2 above)
Comment (LeoLiao123) [reply ray-operator/test/e2eautoscaler/raycluster_autoscaler_test.go]: Fixed.

**Comment by MortalHappiness** [comment]
Please rebase with the master branch. Thanks.

**Pull request closed without merging**